tool/tsh/common/tsh.go

```diff
@@ -705,6 +706,38 @@ func initLogger(cf *CLIConf) {
 //
 // DO NOT RUN TESTS that call Run() in parallel (unless you taken precautions).
 func Run(ctx context.Context, args []string, opts ...CliOption) error {
+	// At process startup, check if a version has already been downloaded to
+	// $TELEPORT_HOME/bin or if the user has set the TELEPORT_TOOLS_VERSION
+	// environment variable. If so, re-exec that version of {tsh, tctl}.
+	toolsDir, err := tools.Dir()
```

Comment:
This broke VNet in v17.0.1. VNet has a launch daemon on macOS which starts `tsh vnet-daemon` in an environment where `$HOME` doesn't exist.

Before 9d2b5c5fe9ec4a6, it was possible to execute `tsh version` with no env vars:

```
env -i "$(which tsh)" version
```

What do you reckon would be the best way forward? I don't want to set some dummy value as `$HOME` because I don't want to interfere with how launch daemons on macOS behave. Setting `TELEPORT_TOOLS_VERSION=off` is not enough as `tools.Dir()` is read before we check that env var.

Reply:
I tried to explicitly provide `TELEPORT_HOME` with a dummy dir so that `tools.Dir()` doesn't need to read `HOME`, but that doesn't work either. Not because of autoupdate, but because I think when you explicitly provide `TELEPORT_HOME`, tsh tries to read the config file which it cannot in this case.

```
~: env -i TELEPORT_HOME=/dev/null TELEPORT_TOOLS_VERSION=off "$(which tsh)" version
ERROR: failed to load tsh config from "/dev/null/config/config.yaml"
        failed to execute command /dev/null/config/config.yaml error:  not a directory
```

Reply:
Thanks, keep me updated. I'll update the plist for the launch daemon with an appropriate env vars once ready.